quranref/cli.py

```diff
@@ -36,48 +33,46 @@ def import_surahs():
             rukus=s_info['rukus'],
             total_ayas=s_info['total_ayas']
         )
-
-        gdb = get_gdb()
         gdb.add(s)
         surah_number += 1
 
 
 @cmd_app.command()
 def import_text(language: str, text_name: str, filename: str):
-    """
-    with typer.progressbar(iteration(), length=1000, label="Progressing") as progress: 
-        for i in progress:
-            sleep(0.01)
-    """
-    file = open(filename)
-
+    """Importing Text""" 
+    file = open('data/'+filename)
     bismillah_text = ''
     current_surah = ''
-
-    for line in file:
-        if not line.strip():
-            break
-
-        surah, aya, content = line.split('|')
-
-        if not bismillah_text:
-            bismillah_text = content.strip()
-
-        if current_surah != surah and content.startswith(bismillah_text):
-            # new surah starting, separate bismillah and save as aya 0 for surah
-            content = content[len(bismillah_text):].strip()
-            if content:
-                aya_doc = Aya.new(surah_number=surah, aya_number=0)
-            else:
-                aya_doc = Aya.new(surah_number=surah, aya_number=aya)
-
-            AyaText.new(aya_doc, bismillah_text, language, text_name)
-            current_surah = surah
-
-        if content:
-            aya_doc = Aya.new(surah_number=surah, aya_number=aya)
-            AyaText.new(aya_doc, content.strip(), language, text_name)
-
+    total = 0
+    file_range = len(file.readlines())
```

Comment:
In order to introduce the progress bar, the logic now reads the entire file at the start. The previous logic was meant so even large files don't require big memory overhead because of the program loading  the full file. In previous implementation it was just one line at a time that was being read from the file.

Reply:
May be to keep the progress bar and avoid the double for loop below we could do something like:

```python
file_lines = file.readlines()
total_lines = len(file_lines)
with typer.progressbar(range(total_lines)) as progress:
    for value in progress:
        line = file_lines[value]
```